modyn/trainer_server/internal/dataset/per_class_online_dataset.py

```diff
@@ -0,0 +1,94 @@
+import gc
+import logging
+from typing import Generator
+
+from modyn.trainer_server.internal.dataset.online_dataset import OnlineDataset
+from torch.utils.data import get_worker_info
+
+logger = logging.getLogger(__name__)
+
+
+class PerClassOnlineDataset(OnlineDataset):
+    # pylint: disable=too-many-instance-attributes, abstract-method
+
+    def __init__(
+        self,
+        pipeline_id: int,
+        trigger_id: int,
+        dataset_id: str,
+        bytes_parser: str,
+        serialized_transforms: list[str],
+        storage_address: str,
+        selector_address: str,
+        training_id: int,
+    ):
+        super().__init__(
+            pipeline_id,
+            trigger_id,
+            dataset_id,
+            bytes_parser,
+            serialized_transforms,
+            storage_address,
+            selector_address,
+            training_id,
+        )
+
+        self._filtered_label = None
```

Comment:
any reason we do not set that in the constructor?

Reply:
Now toy can specify the initial label in the constructor and then change it at runtime